File: src/2020/day6/day6.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
  FILE *file = fopen("2020_day6_smol.txt", "r");
  if (file == NULL) {
    perror("Unable to open file");
    return 1;
  }

  char *line = NULL;
  size_t len = 0;
  ssize_t read;

  while ((read = getline(&line, &len, file)) != -1) {
    line[strcspn(line, "\n")] = 0;

    printf("%s \n", line);
  }

  if (line) {
    free(line);
  }

  fclose(file);
  return 0;
}

File: src/2020/day6/day6.py
file_data= ""

with open("./2020_day6.txt") as f:
	file_data = f.read()
f.close()

groups = file_data.split("\n\n")
s = 0
for group in groups:
	group = group.replace("\n", "")
	s+= (len(set(group)))
print(s)

